### PR TITLE
NFDIV-1717 : BP Notification when respondent has responded without dispute

### PR DESCRIPTION
**Pull request by sunilnatikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1717

### Change description ###
Changes done to generate AoS response (without dispute) letter and sent it to bulk print when applicant is offline.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendAosResponseLetterPackToApplicant.java [hunk 1]
```diff
@@ -23,8 +23,14 @@
         final CaseData caseData = caseDetails.getData();
         final AcknowledgementOfService acknowledgementOfService = caseData.getAcknowledgementOfService();
 
-        if (caseData.getApplicant1().isOffline() && acknowledgementOfService.isDisputed()) {
-            log.info("Sending aos response letter pack to bulk print as applicant1 is offline. Case id: {}", caseId);
+        if (caseData.getApplicant1().isOffline()) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendAosResponseLetterPackToApplicant.java:26]: Are we sure the AoS response only goes to offline applicants?

File: src/main/java/uk/gov/hmcts/divorce/document/content/AosUndefendedResponseLetterTemplateContent.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    private static final String DATE_TO_WAIT_UNTIL_APPLY_FOR_CO = "dateToWaitUntilApplyForCO";
+
+    public Map<String, Object> apply(final CaseData caseData,
+                                     final Long ccdCaseReference) {
+
+        log.info("For ccd case reference {} and type(divorce/dissolution) {} ", ccdCaseReference, caseData.getDivorceOrDissolution());
+
+        Map<String, Object> templateContent = new HashMap<>();
+        templateContent.put(APPLICANT_1_FIRST_NAME, caseData.getApplicant1().getFirstName());
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/document/content/AosUndefendedResponseLetterTemplateContent.java:47]: use Applicant.getFullName instead

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendAosResponseLetterPackToApplicant.java (hunk 1 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/SendAosResponseLetterPackToApplicant.java:26]: Yes Linus, As per the ACs in [NFDIV-1717](https://tools.hmcts.net/jira/browse/NFDIV-1717) and [NFDIV-1718](https://tools.hmcts.net/jira/browse/NFDIV-1718) the AoS response letter is only sent if applicant is flagged as offline.

The email notifications for online applicants are sent separately, as part of task **SendApplicationIssueNotifications**

File: src/main/java/uk/gov/hmcts/divorce/document/content/AosUndefendedResponseLetterTemplateContent.java (hunk 2 above)
Comment (sunilnatikar) [reply src/main/java/uk/gov/hmcts/divorce/document/content/AosUndefendedResponseLetterTemplateContent.java:47]: getFullName returns full name including middle name as well, and verified with Michael and we don't want middle name to be displayed even if it is present that's why set first name and last name separately.

Comment (linusnorton) [reply src/main/java/uk/gov/hmcts/divorce/document/content/AosUndefendedResponseLetterTemplateContent.java:47]: we had an issue raised saying it must be there, it's their full legal name

**Review by linusnorton — APPROVED** [review]

**Pull request merged**